Plugins/Public/base_plugin/PlayerBase.cpp

```diff
@@ -688,23 +731,28 @@ float PlayerBase::SpaceObjDamaged(uint space_obj, uint attacking_space_obj, floa
 				}
 			}
 
-			if (!is_ally)
+			if (!is_ally && ((hostile_tags_damage[charname] + incoming_damage) > damage_threshold))
 			{
-				if (set_plugin_debug > 1)
-					ConPrint(L"PlayerBase::damaged space_obj=%u\n", space_obj);
-				
-				float incoming_damage = abs(curr_hitpoints - new_hitpoints);
-				if (hostile_tags_damage.find(charname) == hostile_tags_damage.end())
-					hostile_tags_damage[charname] = 0;
-
-				if ((hostile_tags_damage[charname] + incoming_damage) < damage_threshold)
-					hostile_tags_damage[charname] += incoming_damage;
-				else hostile_tags[charname] = charname;
-					
+				hostile_tags[charname] = charname;
+
+				const wstring& charname = (const wchar_t*)Players.GetActiveCharacterName(client);
+				ReportAttack(this->basename, charname, this->system, L"has activated self-defense against");
 
 				SyncReputationForBase();
+
+				if (siege_mode)
+					SiegeModChainReaction(client);
 			}
 		}
+
+		if (!siege_mode && ((hostile_tags_damage[charname] + incoming_damage) > siege_mod_damage_trigger_level))
+		{
+			const wstring& charname = (const wchar_t*)Players.GetActiveCharacterName(client);
+			ReportAttack(this->basename, charname, this->system, L"siege mode triggered by");
```

Comment:
I'm beginning to worry about how this commit in general will look to players, we're adding a lot of system-wide messages in this commit

Reply:
I could minify their amount to absolutely necessary one

I thought, the more the merrier. They give a complete understanding of a happening situation